File: src/main/java/icu/samnyan/aqua/sega/chusan/model/userdata/UserCardPrintState.java
package icu.samnyan.aqua.sega.chusan.model.userdata;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Entity(name = "ChusanUserCardPrintState")
@Table(name = "chusan_user_print_state")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserCardPrintState extends Chu3UserEntity {
    private boolean hasCompleted;
    private LocalDateTime limitDate;
    private int placeId;
    private int cardId;
    private int gachaId;
}

File: src/main/java/icu/samnyan/aqua/sega/chusan/model/userdata/UserCardPrintState.kt
package icu.samnyan.aqua.sega.chusan.model.userdata

import jakarta.persistence.Entity
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity(name = "ChusanUserCardPrintState")
@Table(name = "chusan_user_print_state")
class UserCardPrintState : Chu3UserEntity() {
    var hasCompleted = false
    var limitDate: LocalDateTime = LocalDateTime.now()
    var placeId = 0
    var cardId = 0
    var gachaId = 0
}
